src/App.js

```diff
@@ -1,16 +1,85 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
+import ColorChoice from './components/ColorChoice';
 
 const App = () => {
+  const[entriesData, setEntriesData] = useState(chatMessages);
+
+  const likeEntry = (id) => {
+    setEntriesData((prev)=>{
+      return prev.map((entry)=>{
+        if(id === entry.id){
+          return{
+            ...entry,
+            liked: !entry.liked
+          };}
+        else{
+          return entry;
+        }
+        });
+    });
+  };
+
+  const totalLikes= entriesData.reduce((total, entry)=>{
+    if(entry.liked === true){
+      total +=1;
+    }
+    return total;
+  }, 0);
+
+  const [senderColors, setSenderColors] = useState({
+    [chatMessages[0].sender]: 'black',
+    [chatMessages[1].sender]: 'black'
+  });
+
+  const handleSenderColor = (sender, color) => {
+    setSenderColors((prevColors) => ({
+      ...prevColors,
+      [sender]: color
```

Comment:
👍 Great use of the computed property syntax to update the color for the particular sender.